service/environment.go

```diff
@@ -144,6 +147,26 @@ func (e *environment) Instancers() Instancers {
 	return e.instancers.Copy()
 }
 
+func (e *environment) UpdateInstancers(currentKeys map[string]bool, instancersToAdd Instancers) {
+	// add new instancers
+	for key, value := range instancersToAdd {
+		e.instancers.Set(key, value)
+	}
+
+	// remove outdated instancers
+	for key := range e.instancers {
+		if _, ok := currentKeys[key]; !ok {
+			i, found := e.instancers.Get(key)
+
+			if found {
+				i.Stop()
+				delete(e.instancers, key)
```

Comment:
Since the environment returns a copy of the instancers on read https://github.com/xmidt-org/webpa-common/blob/main/service/environment.go#L144 we might run into a data race here. 

We'll have to either synchronize reads/writes or look to see if we could replace the whole instancers map with the new one (in which case we could use https://golang.org/pkg/sync/atomic/#example_Value_config for atomic map swap)